packages/icon/cli/generate-components.js

```diff
@@ -5,34 +5,85 @@ const SVGO = require('svgo')
 
 const svgo = new SVGO({ plugins: [{ removeXMLNS: true }] })
 
-const regex = />/
-const componentTemplate = (svg, componentName, core) => `
-import React from 'react'
-import PropTypes from 'prop-types'
-import Icon from ${
-  core ? "'../index.js'" : "'@pluralsight/ps-design-system-icon'"
-}
-export const ${componentName} =  React.forwardRef(function ${componentName}({'aria-label': ariaLabel, ...props}, ref){ return (
-  <Icon {...props} ref={ref}>
-    ${svg.replace(
-      regex,
-      "role='img' {...(ariaLabel && { 'aria-label': ariaLabel })}>"
-    )}
-  </Icon>
-)})
-${componentName}.displayName = '${componentName}'
-${componentName}.propTypes = {
-  'aria-label': PropTypes.string,
+exports.generateComponents = async ({
```

Comment:
Is this just a file resort?  Or are there some significant changes you would highlight here?

Reply:
the only significant thing was the new component code. everything else was just trying to make sure i left it better and more understandable